src/debug/providerRegistry.ts

```diff
@@ -0,0 +1,40 @@
+import * as vscode from "vscode";
+
+import { IDebugProvider } from "./debugProvider";
+import { JavaDebugProvider } from "./javaDebugProvider";
+
+const supportedProviders: IDebugProvider[] = [
+    new JavaDebugProvider()
+];
+
+async function showProviderPick(): Promise<IDebugProvider> {
+    if (supportedProviders.length < 1) {
+        return null;
+    } else if (supportedProviders.length === 1) {
+        return supportedProviders[0];
+    }
+
+    const providerItems = supportedProviders.map((provider) => {
+        return {
+            label: provider.getDebuggerType(),
+            provider
+        };
+    });
+    
+    const pickedProvider = await vscode.window.showQuickPick(<any[]> providerItems, { placeHolder: "Select the environment" });
+    if (!pickedProvider) {
+        return null;
+    }
+    return pickedProvider.provider;
+}
+
+export async function getDebugProvider(baseImage?: string): Promise<IDebugProvider> {
+    let debugProvider = null;
+    if (baseImage) {
+        debugProvider = supportedProviders.find((provider) => provider.isSupportedImage(baseImage));

```

Comment:
Is it ever possible for a base image to be supported by more than one provider?

Reply:
This's a good concern. This is involved with how intelligent our code can infer the application type. In the `javaDebugProvider#isSupportedImage`, i didn't add too many guess logic, but just consider the classic key words `java`, `openjdk`, `oracle` as a java image. If in such cases, it points to multiple providers, i think it makes sense to the first provider matched by the loop.

Reply:
That sounds fine - we can address that later if it comes up.